src/stores/reactions.js

```diff
@@ -0,0 +1,240 @@
+/**
+ * @copyright Copyright (c) 2023 Dorra Jaouad <dorra.jaoued7@gmail.com>
+ *
+ * @author Marco Ambrosini <marcoambrosini@icloud.com>
+ * @author Dorra Jaouad <dorra.jaoued7@gmail.com>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import { defineStore } from 'pinia'
+import Vue from 'vue'
+
+import { getReactionsDetails } from '../services/messagesService.js'
+
+/**
+ * @typedef {string} Token
+ */
+
+/**
+ * @typedef {number} MessageId
+ */
+
+/**
+ * @typedef {object} Reactions
+ * @property {string} emoji - reaction emoji
+ * @property {object} participant - reacting participant
+ */
+
+/**
+ * @typedef {object} State
+ * @property {{[key: Token]: {[key: MessageId]: Reactions}}} reactions - The reactions per message.
+ */
+
+/**
+ * Store for conversation extra chat features apart from messages
+ *
+ * @param {string} id store name
+ * @param {State} options.state store state structure
+ */
+export const useReactionsStore = defineStore('reactions', {
+	state: () => ({
+		reactions: {},
+	}),
+
+	getters: {
+		getReactions: (state) => (token, messageId) => {
+			return state.reactions?.[token]?.[messageId]
+		},
+	},
+
+	actions: {
+		/**
+		 * Add a reaction for a given message.
+		 *
+		 * @param {object} payload action payload
+		 * @param {string} payload.token The conversation token
+		 * @param {number} payload.messageId The id of message
+		 * @param {string} payload.reaction The reaction to add
+		 * @param {object} payload.actors The users who reacted
+		 *
+		 */
+		addReaction({ token, messageId, reaction, actors }) {
+			if (this.reactions[token]?.[messageId]) {
```

Comment:
both `if` possible outcomes don't covered by test. Don't we need to create a message, if we're adding a reaction?

check `this.checkForExistence` method in sharedItems store, maybe we could replicate it for this store as well